presto-tests/src/test/java/io/prestosql/execution/TestEventListener.java

```diff
@@ -260,14 +281,19 @@ private void assertFailedQuery(Session session, @Language("SQL") String sql, boo
         assertEquals(expectedFailure, failureInfo.getFailureMessage().orElse(null));
     }
 
-    private EventFilters buildEventFilters(boolean ignoreSplits)
+    private EventFilters buildEventFilters(String queryMarker, boolean ignoreSplits)
     {
+        Instant now = Instant.now();
         EventFilters.Builder builder = EventFilters.builder()
-                .setQueryCreatedFilter(event -> !event.getMetadata().getQuery().contains(IGNORE_EVENT_MARKER))
-                .setQueryCompletedFilter(event -> !event.getMetadata().getQuery().contains(IGNORE_EVENT_MARKER));
-        if (ignoreSplits) {
-            builder.setSplitCompletedFilter(event -> false);
-        }
+                .setQueryCreatedFilter(event -> {
+                    boolean queryOk = event.getMetadata().getQuery().contains(queryMarker);
+                    if (!queryOk) {
+                        ignoredQueries.add(event.getMetadata().getQueryId());
+                    }
+                    return queryOk;
+                })
+                .setQueryCompletedFilter(event -> !ignoredQueries.contains(event.getMetadata().getQueryId()))
```

Comment:
why not `event.getMetadata().getQuery().contains(queryMarker)` ?

Reply:
With ignored queries recording mechanism in place it felt natural to only look at the marker for "create" event. If we can get rid of that mechanism (ignoring splits events) then I will get back to using marker.